### PR TITLE
feat: support auto READMEs

### LINKED ISSUE #100: Use URL + /#readme for readme
We could remove the readme links and just append `/#readme` to the `project.url` if `readme:` is missing. It's really only used on one page that likely is not the one most people would use; the main page just has normal links. A link to the package itself should not auto-scroll the page, because users might want a variety of information from visiting a project, like the most recent download, the stars button (very important!), the clone button, the issues tab, etc. I hope everyone knows how to scroll down on GitHub (and would really like it if GitHub made the files collapsible), but I don't want to hide the important links at the top of the page, at least for the "normal" link. On the packages page, linking to the README's is fine, since it's a separate link that says README.

_Originally posted by @henryiii in https://github.com/scikit-hep/scikit-hep.github.io/issues/99#issuecomment-745455755_

### PR DESCRIPTION
**Pull request by henryiii**
Closes #100.

CC @jpivarski, it is based on his idea.

- docs: use main page `/#readme` as README unless a custom readme is specified.
- docs: allow docs title override (Awkward's docs isn't really a ReadTheDocs, but a Website)

### CODE REVIEW

File: _data/projects/basic/awkward-array.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: awkward-array
 projlogo: /assets/images/projlogo/logo_awkward_array.png
 description: Manipulate JSON-like data with NumPy-like idioms.
 url: https://github.com/scikit-hep/awkward-1.0
-readme: https://github.com/scikit-hep/awkward-1.0/blob/main/README.md
 docs: https://awkward-array.org
+docs-title: Website
```
Comment (eduardo-rodrigues) [inline _data/projects/basic/awkward-array.yml:6]: You might as well use "awkward-array.org" as the title?

**Review by eduardo-rodrigues — APPROVED** [review]
Seems all fine to me.

File: _data/projects/basic/awkward-array.yml (hunk 1 above)
Comment (henryiii) [reply _data/projects/basic/awkward-array.yml:6]:
```suggestion
docs-title: awkward-array.org
```

Comment (henryiii) [reply _data/projects/basic/awkward-array.yml:6]: I don’t know, I’m happy to let @jpivarski decide.

Comment (jpivarski) [reply _data/projects/basic/awkward-array.yml:6]: Sure!

**Pull request merged**